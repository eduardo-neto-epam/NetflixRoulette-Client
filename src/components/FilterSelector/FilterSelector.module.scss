.filter {
    font-family: var(--font-main);
    font-size: 24px;
    text-transform: uppercase;
    line-height: 48px;
    color: var(--color-contrast);
    cursor: pointer;
    position: relative;

    .checkmark {
        width: 100%;
        height: 4px;
        position: absolute;
        bottom: -21px;
        left: 0;
        background: red;
        display: none;
    }

    input {
        position: absolute;
        opacity: 0;
        cursor: pointer;
        height: 0;
        width: 0;

        &:checked ~ .checkmark {
            display: block;
        }
    }

    
}


