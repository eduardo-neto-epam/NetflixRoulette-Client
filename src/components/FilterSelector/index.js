import React, { useEffect, useState } from 'react'
import styles from './FilterSelector.module.scss'
import classnames from 'classnames/bind'
import PropTypes from 'prop-types';

let cx = classnames.bind(styles)

const FilterSelector = ({ children, handleSelect }) => {

    return (
        <label className={cx('filter')} htmlFor={children}>{children}
        <input 
            type='radio'
            name='filters'
            id={children}
            defaultChecked={children === 'all'}
            className={cx(
                'menu-filter'
                )}
            onChange={(e) => handleSelect(e)}
            value={children}
        />
         <span className={cx('checkmark')} />
        </label>
    )
}

FilterSelector.propTypes = {
    children: PropTypes.string,
    handleSelect: PropTypes.func,
  }
  
  FilterSelector.defaultProps = {
    children: 'LABEL',
    handleSelect: ()=>{throw new Error('Prop to handle filter selection is missing')},
  }

export default FilterSelector