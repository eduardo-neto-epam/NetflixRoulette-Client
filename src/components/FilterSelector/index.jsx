import React, { useEffect, useState } from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames/bind';
import styles from './FilterSelector.module.scss';

const cx = classnames.bind(styles);

const FilterSelector = ({ filter, children, setFilter }) => {
  const [isChecked, setIsChecked] = useState(false);

  useEffect(() => {
    setIsChecked(filter === children);
  }, [filter]);

  return (
    <label className={cx('filter')} htmlFor={children}>
      {children}
      <input
        type="radio"
        name="filters"
        id={children}
        checked={isChecked}
        className={cx(
          'menu-filter',
        )}
        onChange={(e) => setFilter(e.target.value)}
        value={children}
      />
      <span className={cx('checkmark')} />
    </label>
  );
};

FilterSelector.propTypes = {
  filter: PropTypes.string.isRequired,
  children: PropTypes.string.isRequired,
  setFilter: PropTypes.func.isRequired,
};

export default FilterSelector;
