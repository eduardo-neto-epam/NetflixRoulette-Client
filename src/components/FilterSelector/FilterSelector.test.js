import '@testing-library/react'
import renderer from 'react-test-renderer';
import React from 'react'
import ReactDOM  from 'react-dom'
import { act } from 'react-dom/test-utils';
import { fireEvent, cleanup } from '@testing-library/react';
import FilterSelector from './index'

let container;

beforeEach(() => {
  container = document.createElement('div');
  document.body.appendChild(container);
});

afterEach(() => {
  document.body.removeChild(container);
  container = null;
});

afterEach(cleanup);

test('FilterSelector renders correctly', ()=>{
  const tree = renderer
    .create(<FilterSelector>Filter</FilterSelector>)
    .toJSON();
  expect(tree).toMatchSnapshot();
})

test('FilterSelector renders with text given as child', () => {

    const label = 'Test Filter'

    act(() => { ReactDOM.render(<FilterSelector>{label}</FilterSelector>, container) });

    const filter = container.querySelector('label');

    expect(filter.textContent).toBe(label)
})

test('simulates click', () => {

  const label = 'Test Filter'

  const mockedClick = jest.fn()

  function testClick() {

    act(() => { ReactDOM.render(<FilterSelector handleSelect={mockedClick}>{label}</FilterSelector>, container) });

    let input = container.querySelector('input');

    fireEvent.click(input);

    expect(mockedClick.mock.calls.length).toEqual(1)

  }

  testClick();

})
