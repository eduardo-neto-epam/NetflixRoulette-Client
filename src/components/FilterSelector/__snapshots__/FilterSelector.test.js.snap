// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilterSelector renders correctly 1`] = `
<label
  className="filter"
  htmlFor="Filter"
>
  Filter
  <input
    className="menu-filter"
    defaultChecked={false}
    id="Filter"
    name="filters"
    onChange={[Function]}
    type="radio"
    value="Filter"
  />
  <span
    className="checkmark"
  />
</label>
`;
